backend/deepchecks_monitoring/api/v1/data_ingestion_alert_rule.py

```diff
@@ -0,0 +1,177 @@
+from deepchecks_monitoring.utils import auth
+from deepchecks_monitoring.utils.alerts import AlertSeverity, Condition, Frequency
+
+from .router import router
+
+
+class DataIngestionAlertRuleCreationSchema(BaseModel):
+    """Schema defines the parameters for creating new alert rule."""
+
+    name: str
+    condition: Condition
+    alert_severity: t.Optional[AlertSeverity]
+    is_active: t.Optional[bool]
+    alert_type: AlertRuleType
+    frequency: Frequency
+
+
+class DataIngestionAlertRuleSchema(BaseModel):
+    """Schema for the alert rule."""
+
+    id: int
+    name: str
+    model_id: int
+    condition: Condition
+    alert_severity: t.Optional[AlertSeverity]
+    is_active: bool
+
+    class Config:
+        """Config for Alert schema."""
+
+        orm_mode = True
+
+
+class DataIngestionAlertRuleUpdateSchema(BaseModel):
+    """Schema defines the parameters for updating alert rule."""
+
+    name: t.Optional[str]
+    alert_severity: t.Optional[AlertSeverity]
+    condition: t.Optional[Condition]
+    is_active: t.Optional[bool]
+
+
+@router.post(
+    "/models/{model_id}/data-ingestion-alert-rules",
+    response_model=IdResponse,
+    tags=[Tags.ALERTS],
+    dependencies=[Depends(Model.get_object_from_http_request)],
+    summary="Create new alert rule on a given model."
+)
+async def create_alert_rule(
+        model_id: int,
+        alert_rule: DataIngestionAlertRuleCreationSchema,
+        session: AsyncSession = AsyncSessionDep,
+        user: User = Depends(auth.CurrentUser()),
+):
+    """Create new alert rule on a given check."""
+    stm = insert(DataIngestionAlertRule).values(
+        model_id=model_id,
+        created_by=user.id,
+        updated_by=user.id,
+        **alert_rule.dict(exclude_none=True)
+    ).returning(DataIngestionAlertRule.id)
+
+    rule_id = (await session.execute(stm)).scalar_one()
+    return {"id": rule_id}
+
+
+
+@router.get("/data-ingestion-alert-rules/{data_ingestion_alert_rule_id}",
+            response_model=DataIngestionAlertRuleSchema, tags=[Tags.ALERTS])
+async def get_alert_rule(
+        data_ingestion_alert_rule_id: int,  # pylint: disable=unused-argument
+        alert_rule: DataIngestionAlertRule = Depends(DataIngestionAlertRule.get_object_from_http_request)
+):
+    """Get data-ingestion alert rule by id."""
+    return DataIngestionAlertRuleSchema.from_orm(alert_rule)
+
+
+@router.put("/data-ingestion-alert-rules/{data_ingestion_alert_rule_id}",
+            tags=[Tags.ALERTS],
+            dependencies=[Depends(DataIngestionAlertRule.get_object_from_http_request)],
+            summary="Update data-ingestion alert rule by id.")
+async def update_alert(
+        data_ingestion_alert_rule_id: int,
+        body: DataIngestionAlertRuleUpdateSchema,
+        session: AsyncSession = AsyncSessionDep,
+        user: User = Depends(auth.CurrentUser()),
+):
+    """Update data-ingestion alert rule by id."""
+    updated_body = body.dict(exclude_unset=True).copy()
```

Comment:
I think you also need to add `exclude_defaults`/

example

```python
# %%
import pydantic

class Foo(pydantic.BaseModel):
    a: str | None
    b: str | None

Foo(**{'a': 'a', 'b': None}).dict(exclude_unset=True)
# {'a': 'a', 'b': None}
```

Reply:
there are no defaults

Reply:
the body dict is read only I think

Reply:
It is like that in many endpoints